src/ui/common/src/components/pages/workflow/id/index.tsx

```diff
@@ -116,11 +123,25 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
     }
   }, [workflow.selectedResult, urlSearchParams, navigate]);
 
+  // EFFECT 3: Load workflow metadata.
+  // This useEffect is effectively only called on component mount. It loads
+  // the base workflow metadata as well as metadata about any integrations
+  // in order to populate the UI.
   useEffect(() => {
     dispatch(handleGetWorkflow({ apiKey: user.apiKey, workflowId }));
     dispatch(handleLoadIntegrations({ apiKey: user.apiKey }));
   }, [dispatch, user.apiKey, workflowId]);
 
+  // EFFECT 4: Gather selected workflow index.
+  // When the workflow Redux store's DAG results are populated or when we navigate
+  // to a different version, we iterate through the full list of results and set the
+  // index in both Redux and in our local state.
+  // NOTE(vikram): There are two annoying bits of tech debt in this code:
+  // 1. It's not clear that this needs to be a different from Effect 2 the one where we
```

Comment:
Most likely these can be combined into one useEffect

Reply:
Yep, that was my thinking as well. I didn't have full confidence in my understanding of the nuances of this (or the time to test), so I wanted to document at a minimum before losing the context.